### ISSUE TITLE
Allow specifying multiple SSH keys with `provision -h connect`

### ISSUE THREAD

**Issue by WOnder93** [teemtee/tmt#1491]
TMT internally supports specifying multiple SSH keys for one guest, but the CLI for the `connect` method supports only a single key path.

I would submit a PR, but I'm not sure what would be the preferred way to specify multiple keys. `-k <key1> -k <key2>`? `-k <key1>,<key2>`? `-k <key1>:<key1>`?

**Comment by happz** [comment]
`-k <key1> -k <key2>` should be the lowest hanging fruit, it could be just a question of fixing `-k` option of a plugin to allow multiple values. Click would collect them, and `key` as input for guest creation is already supposed to be a list of strings/keys.

**Comment by WOnder93** [comment]
`-k <key1> -k <key2>` would slightly "break" the command-line API, since currently such invocation is valid and results in the last key being used. This would change to all keys being passed to SSH after the change. It would probably be harmless in practice, as the last key would be included anyway, but I'd like to confirm first if it's acceptable.

**Comment by lukaszachy** [comment]
I'd say the fact that '-k <key1> -k <key2>' is currently accepted is a surprise. So fixing it is the right thing to do.

**Comment by WOnder93** [comment]
Why a surprise? It is a common CLI convention that when a single-value option is specified multiple times that the last value is taken. Try for example `ls -w 20` and `ls -w 20 -w 0`. It is sometimes useful in scripts, when you want to set some value as a default but allow the callers to override it. For example, let's say you have a program `program` with 0 as the default for `--value` and a script `script.sh` with contents:
```sh
#!/bin/sh

program --value 10 "$@"
```

Then calling `./script` will call `program` with value 10 and `./script --value 20` with value 20.

**Comment by lukaszachy** [comment]
Well it was surprised for me. I'm sure it is documented somewhere and have all best intentions. Same as "-all" is in reality "-a -l -l".

**Comment by psss** [comment]
I don't think we would break anyone (or at least not many users) who would be relying on the last key. Let's support multiple keys. Adding to the backlog.

**Comment by psss** [comment]
Should be fixed by #4147.
